lib/opencensus/trace/exporters/ocagent.rb

```diff
@@ -15,14 +15,263 @@
+          elsif credentials.is_a? String
+            @credentials = GRPC::Core::ChannelCredentials.new credentials
+          end
+
+          @node_info = create_node_info
+          @resource = create_resource(
+            resource_type || GLOBAL_RESOURCE_TYPE,
+            labels: resource_lables
+          )
+          @trace_stream_sleep_delay =
+            trace_stream_sleep_delay || TRACE_STREAM_SLEEP_DELAY
+          @stopped = true
+          @trace_req_queue = nil
+        end
+
+        # Create the OpenCensus Agent trace service grpc client.
+        # @return [OpenCensus::Proto::Agent::Trace::V1::TraceService::Stub]
+        def client
+          @client ||= AgentProto::Trace::V1::TraceService::Stub.new(
+            agent_service_address,
+            credentials
+          )
+        end
+
+        # Create trace config proto.
+        # Global configuration of the trace service.
+        #
+        # @param [Integer] max_attributes The max number of
+        #   attributes per span. Optional
+        # @param [Integer] max_annotations The max number of
+        #   annotation events per span. Optional
+        # @param [Integer] max_message_events The max number of
+        #   message events per span. Optional
+        # @param [Integer] max_linnks The global max number of link
+        #   entries per span. Optional
+        # @param [OpenCensus::Trace::Exporters::Sampler, nil] sampler
+        #   The sampler used to make decisions on span sampling.
+        #   See {Sampler} for different type of samplers.
+        # @return [OpenCensus::Proto::Trace::V1::TraceConfig]
+        #
+        def create_trace_config \
+            max_attributes: nil,
+            max_annotations: nil,
+            max_message_events: nil,
+            max_linnks: nil,
+            sampler: nil
+          options = {
+            max_number_of_attributes: max_attributes,
+            max_number_of_annotations: max_annotations,
+            max_number_of_message_events: max_message_events,
+            max_number_of_links: max_linnks
+          }
+
+          sampler_proto = sampler ? sampler.to_proto : nil
+
+          case sampler_proto
+          when OpenCensus::Proto::Trace::V1::ProbabilitySampler
+            options[:probability_sampler] = sampler_proto
+          when OpenCensus::Proto::Trace::V1::ConstantSampler
+            options[:constant_sampler] = sampler_proto
+          when OpenCensus::Proto::Trace::V1::RateLimitingSampler
+            options[:rate_limiting_sampler] = sampler_proto
+          end
+
+          options.delete_if!(&:nil?)
+          OpenCensus::Proto::Trace::V1::TraceConfig.new options
+        end
+
+        # Stop the trace export stream.
+        def stop
+          @stopped = true
+          @trace_req_queue&.push TraceEnumeratorQueue::SENTINEL
+        end
+
+        def stopped?
+          @stopped
+        end
+
+        # Start trace export stream
+        def start
+          return if @trace_req_queue
```

Comment:
What is the usage intent here? Can users repeatedly start - stop - then start again? (It looks like the current code wouldn't allow that due to this line.)

Reply:
yes got your point. Make change according to stopped flag.